### PR TITLE
Run aws and azure tests in parallel.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,3 +13,11 @@ travis_cron: all
 travis_push: only_build only_test_fast publish_packages
 travis_pull_request: only_build only_test_fast
 travis_api: all
+
+TESTPARALLELISM := 20
+
+test_all::
+	PATH=$(PULUMI_BIN):$(PATH) go test -v -count=1 -cover -timeout 2h -parallel ${TESTPARALLELISM} .
```
Comment (lukehoban) [inline Makefile]: Couldn't we have used `./...` or similar to run all nested tests?  That would presumably avoid the need for making the nested tests not real Go tests and the need for the harness at top level.  The individual tests already mark themselves as parallel, so I don't think we need to inject that via the `cloud_test.go`.

Basically, I would have thought that we could do just the `Makefile` parts of this change along with `./...` in this `Makefile`?

Comment (CyrusNajmabadi) [reply Makefile]: the primary problem that @ellismg identified is that if you do that, then go will think of these as two separate sets of tests, and you can end up in hte situation where the go test harness is running one, but emitting no output.  This then causes travis to kill the job.

Comment (lukehoban) [reply Makefile]: Huh - that seems strange - I would have thought that was the intended way of running tests spread throughout a project like this?  @ellismg curious what the issue is that prevents this from working for us?

It's just a shame to not be able to make the individual tests actual go tests (prevents easily locally testing in a single provider for example).

Comment (CyrusNajmabadi) [reply Makefile]: I'll wait on @ellismg .  I may have misunderstood things.

Comment (ellismg) [reply Makefile]: The issue is that when go test runs against multiple packages, it buffers all the output from running a packge’s tests (even when you explicitly pass -v) until testing of the package completes.

We rely on the incremental output to appease Travis’s hurstistics where if a build does not product output for 10 mins, it is considered to have hung and the job is killed.

**Pull request merged**